openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -330,14 +345,54 @@ def test_disable_credit_requirements(self):
     def test_is_user_eligible_for_credit(self):
         credit_course = self.add_credit_course()
         CreditEligibility.objects.create(
-            course=credit_course, username="staff"
+            course=credit_course, username=self.user.username
         )
-        is_eligible = api.is_user_eligible_for_credit('staff', credit_course.course_key)
+        is_eligible = api.is_user_eligible_for_credit(self.user.username, credit_course.course_key)
         self.assertTrue(is_eligible)
 
         is_eligible = api.is_user_eligible_for_credit('abc', credit_course.course_key)
         self.assertFalse(is_eligible)
 
+    @ddt.data(
+        CourseMode.AUDIT,
+        CourseMode.HONOR,
+        CourseMode.CREDIT_MODE
```

Comment:
(note this is not a blocker to merge this, just generally a little worried about assuming this)